src/HotChocolate/Core/src/Types/Types/Descriptors/ObjectTypeDescriptor.cs

```diff
@@ -69,6 +69,19 @@ protected ObjectTypeDescriptor(IDescriptorContext context)
                 Definition.AttributesAreApplied = true;
             }
 
+            foreach (ObjectFieldDescriptor field in Fields)
+            {
+                if (!field.Definition.Ignore)
+                    continue;
+
+                // if this definition is used for a type extension we need a 
+                // binding to a field which shall be ignored. In case this is a 
+                // definition for the type it will be ignored by the type initialization.
+                Definition.FieldIgnores.Add(new ObjectFieldBinding(
```

Comment:
I have added a note why we have this binding here so that in case of a refactoring we know why this is done.